src/store/transactions/selectors.ts

```diff
@@ -19,3 +20,15 @@ export const selectTransactionLastTxId = createSelector(
   selectTransactionsSlice,
   state => state.lastTxId
 );
+
+export const selectOriginContractSource = (contractName: string) =>
+  createSelector(selectTransactionsSlice, state => {
+    const address = contractName.split('.')[0];
+    const txs = Object.values(state.entities) as Transaction[];
+
+    const contract = txs.find(
+      (tx: Transaction) => tx.tx_type === 'smart_contract' && tx.sender_address === address
+    ) as SmartContractTransaction;
```

Comment:
Should now be `SmartContractTransaction` implictly.

```suggestion
    const contract = txs.find(
      (tx: Transaction) => tx.tx_type === 'smart_contract' && tx.sender_address === address
    );
```